internal/isoeditor/rhcos.go

```diff
@@ -326,3 +371,51 @@ func IgnitionImageArchive(ignitionConfig string) ([]byte, error) {
 
 	return compressedBuffer.Bytes(), nil
 }
+
+// Writing the offsets of initrd images in the end of system area (first 32KB).
+// As the ISO template is generated by us, we know that this area should be empty.
+func writeHeader(ignitionOffsetInfo, ramDiskOffsetInfo *OffsetInfo, isoPath string) error {
+	iso, err := os.OpenFile(isoPath, os.O_WRONLY, 0o664)
+	if err != nil {
+		return err
+	}
+	defer iso.Close()
+
+	// Starting to write from the end of the system area in order to easily support
+	// additional offsets (and as done in coreos-assembler/src/cmd-buildextend-live)
+	headerEndOffset := int64(32768)
+
+	// Write ignition config
+	headerEndOffset, err = writeOffsetInfo(headerEndOffset, ignitionOffsetInfo, iso)
+	if err != nil {
+		return err
+	}
+
+	// Write ram disk
+	_, err = writeOffsetInfo(headerEndOffset, ramDiskOffsetInfo, iso)
```

Comment:
At a glance this looks like it would overwrite the ignition info with the ramdisk info. I think making the change suggested about the return value of `writeOffsetInfo` will make this easier to understand.

Reply:
used 'writtenBytesLength' returned from 'writeOffsetInfo', so the offset calc is made here instead.